kube/src/config/mod.rs

```diff
@@ -7,36 +7,39 @@
 
 mod apis;
 mod exec;
-mod incluster_config;
-mod kube_config;
-mod utils;
+pub mod incluster_config;
+pub(crate) mod kube_config;
+pub(crate) mod utils;
 
-use crate::{config::kube_config::Der, Error, Result};
-use reqwest::{header, Client, ClientBuilder};
-use std::convert::TryInto;
+use crate::config::{
+    self,
+    kube_config::{Der, KubeConfigOptions},
+};
+use crate::{Error, Result};
 
-use self::kube_config::ConfigLoader;
+pub use self::kube_config::ConfigLoader;
 
-/// Configuration stores kubernetes path and client for requests.
+/// Stores kubernetes cluster url and default namespace.
 #[derive(Clone, Debug)]
-pub struct Configuration {
-    pub base_path: String,
-    pub client: Client,
-
+pub struct Config {
```

Comment:
almost feels like this struct doesn't need to exist anymore if we put the cluster_url and default_namespace inside `ClientConfig`. Then a lot of this falls away as well.

- ClientConfig can implement TryFrom<KubeConfig> for local dev
- ClientConfig can implement try_default, or keep a explicit ctor for incluster